vendors/swedbank-pay-core-library/src/SwedbankPay/Core/Library/Methods/Card.php

```diff
@@ -0,0 +1,202 @@
+                    'callbackUrl' => $urls->getCallbackUrl(),
+                    'termsOfServiceUrl' => $this->configuration->getTermsUrl()
+                ],
+                'payeeInfo' => $this->getPayeeInfo($orderId)->toArray(),
+                'riskIndicator' => $this->getRiskIndicator($orderId)->toArray(),
+                'cardholder' => $order->getCardHolderInformation(),
+                'creditCard' => [
+                    'rejectCreditCards' => $this->configuration->getRejectCreditCards(),
+                    'rejectDebitCards' => $this->configuration->getRejectDebitCards(),
+                    'rejectConsumerCards' => $this->configuration->getRejectConsumerCards(),
+                    'rejectCorporateCards' => $this->configuration->getRejectCorporateCards()
+                ],
+                'metadata' => [
+                    'order_id' => $order->getOrderId()
+                ],
+            ]
+        ];
+
+        try {
+            $result = $this->request('POST', '/psp/creditcard/payments', $params);
+        } catch (\Exception $e) {
+            $this->log(LogLevel::DEBUG, sprintf('%s::%s: API Exception: %s', __CLASS__, __METHOD__, $e->getMessage()));
+
+            throw new Exception($e->getMessage());
+        }
+
+        return $result;
+    }
+
+    /**
+     * Initiate a CreditCard Recurrent Payment
+     *
+     * @param mixed $orderId
+     * @param string $recurrenceToken
+     * @param string|null $paymentToken
+     *
+     * @return Response
+     * @throws \Exception
+     */
+    public function initiateCreditCardRecur($orderId, $recurrenceToken, $paymentToken = null)
+    {
+        /** @var Order $order */
+        $order = $this->getOrder($orderId);
+
+        $params = [
+            'payment' => [
+                'operation' => self::OPERATION_RECUR,
+                'intent' => $this->configuration->getAutoCapture() ? self::INTENT_AUTOCAPTURE : self::INTENT_AUTHORIZATION,
+                'currency' => $order->getCurrency(),
+                'amount' => $order->getAmountInCents(),
+                'vatAmount' => $order->getVatAmountInCents(),
+                'description' => $order->getDescription(),
+                'payerReference' => $order->getPayerReference(),
+                'userAgent' => $order->getHttpUserAgent(),
+                'language' => $order->getLanguage(),
+                'urls' => [
+                    'callbackUrl' => $this->getPlatformUrls($orderId)->getCallbackUrl()
+                ],
+                'payeeInfo' => $this->getPayeeInfo($orderId)->toArray(),
+                'riskIndicator' => $this->getRiskIndicator($orderId)->toArray(),
+                'metadata' => [
+                    'order_id' => $orderId
+                ],
+            ]
+        ];
+
+        // Use Recurrence Token if it's exist
+        if (!empty($recurrenceToken)) {
+            $params['payment']['recurrenceToken'] = $recurrenceToken;
+        } else {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method initiateCreditCardRecur uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)